xocto/ranges.py

```diff
@@ -1054,14 +1054,15 @@ def get_finite_datetime_ranges_from_timestamps(
 
 def any_overlapping(ranges: Iterable[Range[T]]) -> bool:
     """Return true if any of the passed Ranges are overlapping."""
-    ranges = list(ranges)
-    if not ranges:
-        return False
-    range_set = RangeSet[T]([ranges[0]])
-    for range in ranges[1:]:
-        if range_set.intersection(range):
-            return True
-        range_set.add(range)
+    stack: list[Range[T]] = []
+    for range in sorted(ranges):
+        if len(stack) == 0:
+            stack.append(range)
+        else:
+            if stack[-1].intersection(range):
+                return True
+            else:
+                stack.append(range)
```

Comment:
We don't need to keep a stack as we're only looking at the last item:

```suggestion
    prev_range: Range[T] | None = None
    for range in sorted(ranges):
        if prev_range is not None and prev_range & range:
            return True
        prev_range = range
```

Reply:
True! That even helped the performance a little more 👍